lib/controller.js

```diff
@@ -122,6 +124,35 @@ export class AdvancedOpenFileController {
         this.openPath(path);
     }
 
+    openNewFile(path) {
```

Comment:
Instead of moving the entire "open new file" flow into a helper, what if we move the "create a new file" bit only? Then, we could add the confirmation bit to `openPath` and reuse it for both the "new directory" and "new file" flows, if that makes sense? Something like:

```js
if (path.exists()) {
  // blah
} else {
  promptConfirm();
  if (shouldCreateFile) {
    createFile();
  } else {
    createDirectory();
  }
}
```